mlflow/utils/autologging/__init__.py

```diff
@@ -110,13 +110,10 @@ def _update_wrapper_extended(wrapper, wrapped):
     return updated_wrapper
```

Comment:
Move `mlflow/utils/autologging_utils.py` to `mlflow/utils/autologging/__init__.py`, allowing us to add autologging utility modules as separate file(s) in the new `mlflow/utils/autologging` directory in the future.

Also address a TODO in the docstring of the `wrap_patch` function.